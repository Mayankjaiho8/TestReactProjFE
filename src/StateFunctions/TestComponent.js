import React, { Component } from 'react';

class TestComponent extends Component {

        constructor(){
            super();
            this.state = {
                clicked : false
            }
            this.changeDOM = this.changeDOM.bind(this);
        }

        changeDOM(){
            //e.target.value = 'InnerHTML Value changed for '
            this.setState(prevState => {return {clicked: !prevState.clicked}})
        }

        render(){
            const { input } = this.props;
            const { clicked } = this.state;
            const DEFAULT_TEXT = 'Hello..... This is Test Component..... Input ->' + input;
            const CLICKED_TEXT = 'This is clicked';

            return (
                <div id = "testDiv" onClick={ this.changeDOM }>
                    { !clicked ? DEFAULT_TEXT : CLICKED_TEXT}
                </div>
            )
        }        
}

export default TestComponent;