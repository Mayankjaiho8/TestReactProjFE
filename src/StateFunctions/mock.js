export const getPromise1 = () => {
    return new Promise((res, rej) => {
        setTimeout(() => {res('This is promise 1 Response')}, 4000)
    })
}