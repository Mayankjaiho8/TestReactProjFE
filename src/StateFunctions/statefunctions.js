export const arr = ['root', 'carrot', 'mango', 'apple'];

export function add(a,b){
    if(!a || !b){
        return null
    }

    return a+b;
}

export const arrayAdd = (item) => {
    if(!item){
        return null
    }

    arr.push(item);
}

export const drinkAll = (drink, flavor) => {
    if(flavor === 'orange'){
        drink('orange')
    }
}

export const drink = () => {
    console.log('orange flavor is good')
}

export const returningFn = (val)=> {
    if(val <= 5){
        return 'success';
    }
    throw 'error';
}