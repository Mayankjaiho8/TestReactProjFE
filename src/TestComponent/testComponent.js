import React from 'react';

import './../sass/test.scss';

const TestComponent = props => {

    return (
        
        <div className="container">
            Hello Wassup !! This is Test Component

            <div className="nav-bar">
                <a href="">Link 1</a>
                <a href="">Link 2</a>
                <a href="">Link 3</a>
                <a href="">Link 4</a>
            </div>

            <h1>This is H1</h1>
        </div>

    )
}

export default TestComponent;