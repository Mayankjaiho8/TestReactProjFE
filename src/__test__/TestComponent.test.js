import TestComponent from './../StateFunctions/TestComponent';
import { getPromise1 } from './../StateFunctions/mock';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Snapshot', ()=>{
    it('test the Test Component', ()=>{
        const tree = renderer.create(<TestComponent input = { 'Input'} />).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('testing the mock function', ()=> {
    test('test the GetPromise1 function', ()=>{
        return expect(getPromise1()).resolves.toBe('This is promise 1 Response')
    })
})

describe('testing the DOM Manipulation', ()=> {
    let testWrapper;
    beforeEach(() => {
        testWrapper = shallow(<TestComponent input = {'Input'}/>)      
    })

    const DEFAULT_TEXT = 'Hello..... This is Test Component..... Input ->Input';
    const CLICKED_TEXT = 'This is clicked';

    test('test the testDiv DOM', () => {
        expect(testWrapper.text()).toBe(DEFAULT_TEXT);
    })

    test('test the DOM click', () => {
        testWrapper.simulate('click');
        expect(testWrapper.text()).toBe(CLICKED_TEXT);
        testWrapper.simulate('click');
        expect(testWrapper.text()).toBe(DEFAULT_TEXT);
    })
})
