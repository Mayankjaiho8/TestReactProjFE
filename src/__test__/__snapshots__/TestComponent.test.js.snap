// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot test the Test Component 1`] = `
<div
  id="testDiv"
  onClick={[Function]}
>
  Hello..... This is Test Component..... Input -&gt;Input
</div>
`;
